### PR TITLE
Added compression of Davis images and masks by param

### PR DESCRIPTION
**Pull request by coopersigrist**
the weight and height params on davis.py specify the size to compress the data, it will create a new root directory so that it may be reused. 

This may be an issue if mishandled as it can create an infinite number of directories, but that won't happen unless it is misused.

### CODE REVIEW

File: bindsnet/datasets/davis.py [hunk 1]
```diff
@@ -112,25 +136,63 @@ def __init__(
...
-        for seq in sequences_names:
+        for seq in self.sequences_names:
             self.enum_sequences.append(self.sequences[seq])
 
     def __len__(self):
         """
-        calculates the number of sequences the dataset holds
+        Calculates the number of sequences the dataset holds
 
         :Returns the number of sequences in the dataset
```
Comment (djsaunde) [inline bindsnet/datasets/davis.py]: Should have syntax :return: Number of sequences in the dataset.

File: bindsnet/datasets/davis.py [hunk 2]
```diff
@@ -57,6 +60,10 @@ def __init__(
         self.task = task
         self.subset = subset
         self.resolution = resolution
+        self.size = size
+
+        # Sets the boolean converted if the size of the images must be scaled down
+        self.converted = not (self.size == (600, 480))
```
Comment (djsaunde) [inline bindsnet/datasets/davis.py]: Nitpick: can be `self.converted = not self.size == (600, 480)`. Make sure you've installed the black pre-commit hook.

File: bindsnet/datasets/davis.py [hunk 3]
```diff
@@ -112,25 +136,63 @@ def __init__(
...
+        Calculates the number of sequences the dataset holds
 
         :Returns the number of sequences in the dataset
         """
 
         return len(self.sequences)
 
+    def _convert_sequences(self):
+        """
+        Creates a new root for the dataset to be converted and placed into, then copies each image and mask into the given size and stores correctly
```
Comment (djsaunde) [inline bindsnet/datasets/davis.py]: Nitpick: add period at the end of sentence.

File: bindsnet/datasets/davis.py [hunk 4]
```diff
@@ -112,25 +136,63 @@ def __init__(
...
+        """
+
```
Comment (djsaunde) [inline bindsnet/datasets/davis.py]: Delete blank line.

File: bindsnet/datasets/davis.py [hunk 5]
```diff
@@ -112,25 +136,63 @@ def __init__(
...
+                    os.path.join(
+                        self.converted_mask_path, seq, str(ind).zfill(5) + ".png"
+                    )
+                )
+
     def _check_directories(self):
         """
         Verifies that the correct dataset is downloaded; downloads if it isn't and download=True
 
         :Throws FileNotFoundError if the subset sequence, annotation or root folder is missing
```
Comment (djsaunde) [inline bindsnet/datasets/davis.py]: Should be `:raises: ...`

File: examples/datasets/Davis.py [hunk 6]
```diff
@@ -24,9 +24,38 @@
     plot_voltages,
```
Comment (djsaunde) [inline examples/datasets/Davis.py]: Change `Davis.py` -> `davis.py` to follow PEP8 naming conventions.

File: examples/mnist/eth_mnist.py [hunk 7]
```diff
@@ -224,26 +224,26 @@
...
-            spikes_ = {layer: spikes[layer].get("s") for layer in spikes}
-            voltages = {"Ae": exc_voltages, "Ai": inh_voltages}
-
-            inpt_axes, inpt_ims = plot_input(
-                image, inpt, label=labels[step], axes=inpt_axes, ims=inpt_ims
-            )
-            spike_ims, spike_axes = plot_spikes(spikes_, ims=spike_ims, axes=spike_axes)
-            weights_im = plot_weights(square_weights, im=weights_im)
-            assigns_im = plot_assignments(square_assignments, im=assigns_im)
+            # image = batch["image"].view(28, 28)
```
Comment (djsaunde) [inline examples/mnist/eth_mnist.py]: Uncomment plotting logic when you push code.

File: examples/datasets/Davis.py (hunk 6 above)
Comment (coopersigrist) [reply examples/datasets/Davis.py]: hmm, it's lowercase on my machine, I changed it last time I pushed. I'll check that out

File: bindsnet/datasets/davis.py (hunk 2 above)
Comment (coopersigrist) [reply bindsnet/datasets/davis.py]: I do have the hook installed, strange if it's suppose to catch this

**Pull request merged**